api/utils.js

```diff
@@ -38,8 +38,54 @@ module.exports = {
       res.status(403).send("Unauthorized");
     }
   },
+  requireVolunteer: function(req, res, next) {
+    /* Read user from database, and make sure it is in fact an volunteer */
+    try {
+      models.HackerProfile.findAll({
+        where: {
+          userId: req.user.id
+        }
+      }).then(hackerProfiles => {
+        if (hackerProfiles.length >= 1) {
+          if (hackerProfiles[0].get("role") === "volunteer") {
+            return next();
+          } else {
+            res.status(403).send("Unauthorized: Incorrect role");
+          }
+        } else {
+          res.status(403).send("Unauthorized; profile not found");
```

Comment:
I know its bad, but I think we are using 400 to present user facing errors, so just change it so it works w Jack's client sdk

Reply:
Ah right